Buffs revenants a whole bunch so they're less terrible to play

ChangelingRain opened the pull request:
List of changes;

> health is now 3 times essence, from 2
> you can't punch them to death when revealed
> regenerates 2 essence every Life(), from 1
> destroyed revenants respawn in 1 minute, from 3, if the dust is untouched
> harvest only lists mobs within a tile, as it can only harvest in that range anyway
> you can harvest crit mobs again
> harvest gives more essence; 10 for dead mobs, 20 for crit or with a ckey, and 30 for crit and ckey
> overload light and defile both cost 20 energy to activate
> overload light and defile no longer lock you in place for five years while the crew murders you
> overload light costs 30 to unlock
> defile costs 40 to unlock
> defile stuns robots, does more stamina damage to humans
> defile onehits normal windows, twohits reinforced windows
> event triggers, these things will spawn occasionally

ChangelingRain committed:
Merge pull request #1 from tgstation/master

guide to fixing conflicts

ChangelingRain committed:
Buffs the fuck out of revenants. Also cleans some stuff up a little.

Anonus commented on code/modules/mob/living/simple_animal/revenant/revenant.dm:
```diff
@@ -45,11 +44,13 @@
+	if(essence_regenerating && !inhibited && essence < essence_regen_cap) //While inhibited, essence will not regenerate
+		essence += essence_regen_amount
+		if(essence > essence_regen_cap)
+			essence = essence_regen_cap
```
This won't actually fuck up and reduce your 400 essence to 25, because this bit only gets called if your essence is below 25.

Anonus commented:
They're still admin-only for now but that's easy enough to change.
Also @Xhuis I'm totally hijacking your shit sorry m8

tedward1337 commented:
Looking good

Boredone commented:
Minor request. It would be nice if it could emag the other bots too instead of just the janibots and securitrons.
Though I think its just those two as the others can kill maybe?

Anonus commented:
It actually couldn't emag bots before because it didn't open them first and bot emagging is absolute dongs. So now it does.

ChangelingRain committed:
Defile now actually emags bots properly. The code for bot emagging is dongs.

Anonus commented:
Easy enough to do.
As a bonus revenants are now immune to space wind and defile is better at window murdering.

ChangelingRain committed:
Revenants no longer get spacewinded, defile murders the shit out of windows(two casts to kill rwindows, from like 6)

ChangelingRain committed:
Revenant "steal essence" objective now actually possible without doing literally nothing, also no longer runtimes. It was doubly impossible, really.

ChangelingRain committed:
Revenant event now triggers. We're goin for it, we are.
I don't actually know if the event works properly concerning triggering with too few dead mobs but it should work in most cases so ¯\ _(ツ) _/¯

WJohn commented:
This really needs a changelog.

ChangelingRain committed:
General terms changelog for all this.

Xhuis commented:
Seems fine to me :+1:

Dorsisdwarf commented:
Can you make Life Drain not lock you in place? I've never seen a revenant sucessfully eat a body because they're visible, valid, and helpless for like 20-30 seconds. Imagine if pulling out your armblade stunned you for ten seconds every time.

Anonus commented:
I could, but I'd rather make it shorter rather than "start this, run away and never care"
Also it's like 6 seconds with this, or so.

Shadowlight213 commented:
Harvest now works on crit people? I remember it was nerfed to ONLY work on the dead.

Anonus commented:
Yes, it now works on crit people.

RemieRichards commented on code/modules/mob/living/simple_animal/revenant/revenant.dm:
```diff
@@ -215,17 +222,17 @@
 
 /datum/objective/revenant/check_completion()
-	if(!istype(owner.current, /mob/living/simple_animal/revenant) || !owner.current)
+	if(!istype(owner, /mob/living/simple_animal/revenant) || !owner)
```
No.
```owner``` is a ```/datum/mind```, never a revenant
the check for ```owner.current``` was correct.

also remove the ```||``` part of this check, everything after the ```||``` is redundant.

RemieRichards commented:
Temp DNMing due to a pretty big issue with the code.

ChangelingRain committed:
PROPER FIX: CHECK
MAX(0, ESSENCE): CHECK
